pkg/sql/drop_test.go

```diff
@@ -916,6 +916,133 @@ CREATE TABLE t.kv (k CHAR PRIMARY KEY, v CHAR);
 
 }
 
+func TestDropTableUnsplitManuallySplitRanges(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+	params, _ := tests.CreateTestServerParams()
+
+	defer gcjob.SetSmallMaxGCIntervalForTest()()
+
+	s, sqlDB, kvDB := serverutils.StartServer(t, params)
+	defer s.Stopper().Stop(context.Background())
+	ctx := context.Background()
+
+	// Disable strict GC TTL enforcement because we're going to shove a zero-value
+	// TTL into the system with AddImmediateGCZoneConfig.
+	defer sqltestutils.DisableGCTTLStrictEnforcement(t, sqlDB)()
+
+	const numRows = 2*row.TableTruncateChunkSize + 1
+	const numKeys = 3 * numRows
+
+	tableName := "test1"
+	if err := tests.CreateKVTable(sqlDB, tableName, numRows); err != nil {
+		t.Fatal(err)
+	}
+
+	var tableDesc = catalogkv.TestingGetTableDescriptor(kvDB, keys.SystemSQLCodec, "t", tableName)
+
+	nameKey := catalogkeys.MakePublicObjectNameKey(keys.SystemSQLCodec, keys.MinNonPredefinedUserDescID, tableName)
+	gr, err := kvDB.Get(ctx, nameKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !gr.Exists() {
+		t.Fatalf("Name entry %q does not exist", nameKey)
+	}
+
+	tableSpan := tableDesc.TableSpan(keys.SystemSQLCodec)
+	metaStartKey := keys.RangeMetaKey(keys.MustAddr(tableSpan.Key))
+	metaEndKey := keys.RangeMetaKey(keys.MustAddr(tableSpan.EndKey))
+
+	tests.CheckKeyCount(t, kvDB, tableSpan, numKeys)
+
+	// The closure checks whether there is any range has sticky bit
+	// hasManualSplitRange==true: fails when no ranges have sticky bit
+	// hasManualSplitRange==false: fails when any range has sticky bit
+	checkManuallySplitRange := func(hasManualSplitRange bool) {
+		ranges, err := kvDB.Scan(ctx, metaStartKey, metaEndKey, 0)
+		if err != nil {
+			t.Fatal(err)
+		}
+		require.NotEmpty(t, ranges)
+		hasManuallySplitRange := false
+		for _, r := range ranges {
+			var desc roachpb.RangeDescriptor
+			if err := r.ValueProto(&desc); err != nil {
+				t.Fatal(err)
+			}
+			hasManuallySplitRange = hasManuallySplitRange || !desc.GetStickyBit().IsEmpty()
+		}
+		require.Equal(t, hasManuallySplitRange, hasManualSplitRange)
+	}
+
+	// assert that all ranges are not manually split
+	checkManuallySplitRange(false)
```

Comment:
nit: style: when passing `nil` or boolean literals, we like to follow them with a `/* */` comment with the argument name. So, here, `checkManuallySplitRange(false /* hasManualSplitRange */)`.

It sounds like a dumb rule, or at least I thought it was dumb because the IDE tells me the same thing, but I've since found it useful when reviewing.

In this case you're probably better off redefining this function as a predicate and doing `require.True` or `require.False` assertions on the return value.